### PR TITLE
Raise steps limit

### LINKED ISSUE #1652: Raising the limit of steps
### Which component is your feature related to?

Forge

### Feature Request

During one test suite run, we could raise the step limit on demand

If a user overrides the defaults, we should warn if the limit is raised beyond the maximum acceptable in normal conditions (on starknet).

### PR DESCRIPTION
**Pull request by Draggu**
Closes #1652

## Introduced changes

- Added option `--max-n-steps` that allow changing steps limit (3_000_000, now default)

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/runtime/src/starknet/context.rs [hunk 1]
```diff
@@ -140,6 +143,13 @@ pub fn build_context(block_info: BlockInfo) -> EntryPointExecutionContext {
     .unwrap()
 }
 
+pub fn set_max_steps(entry_point_ctx: &mut EntryPointExecutionContext, max_n_steps: u32) {
+    entry_point_ctx.block_context.invoke_tx_max_n_steps = max_n_steps;
```
Comment (Arcticae) [inline crates/runtime/src/starknet/context.rs:147]: Why is this one needed? isn't `vm_run_resources` sufficient?

Comment (Draggu) [reply crates/runtime/src/starknet/context.rs:147]: I followed the logic of `build_context()` -> `build_block_context()` to get same state that with `BlockContext` direct construction.
But yes, in fact it's not necessary.

**Review by Arcticae — APPROVED** [review]

File: docs/src/appendix/snforge/test.md [hunk 2]
```diff
@@ -58,6 +58,10 @@ Display additional info about used resources for passed tests.
 
 ## `--save-trace-data`
 
+Number of maximum steps during single test
```
Comment (piotmag769) [inline docs/src/appendix/snforge/test.md:61]:
```suggestion
Number of maximum steps during a single test. For fuzz tests this value is applied to each subtest separately.
```

File: crates/forge/tests/data/steps/src/lib.cairo [hunk 3]
```diff
@@ -0,0 +1,60 @@
+#[cfg(test)]
+mod tests {
+    #[test]
+    // require 570031 steps
```
Comment (piotmag769) [inline crates/forge/tests/data/steps/src/lib.cairo:4]:
```suggestion
    // requires 570031 steps
```

File: crates/forge/src/main.rs [hunk 4]
```diff
@@ -108,6 +108,10 @@ struct TestArgs {
     /// Save execution traces of all test which have passed and are not fuzz tests
     #[arg(long)]
     save_trace_data: bool,
+
+    /// Max steps per test
+    #[arg(long)]
+    max_n_steps: Option<u32>,
```
Comment (piotmag769) [inline crates/forge/src/main.rs:114]: Can we have the same description as in docs? Clap parses this comment and it is shown in CLI if you type `snforge --help`

File: crates/forge-runner/src/lib.rs [hunk 5]
```diff
@@ -46,18 +45,16 @@ mod running;
...
-        "RangeCheck",
-        "Bitwise",
-        "EcOp",
-        "Poseidon",
-        "SegmentArena",
-        "GasBuiltin",
-        "System",
-    ]
-});
+pub const BUILTINS: [&str; 8] = [
```
Comment (piotmag769) [inline crates/forge-runner/src/lib.rs:48]: Good catch xD Was used as vector before prob

File: crates/runtime/src/starknet/context.rs (hunk 1 above)
Comment (piotmag769) [reply crates/runtime/src/starknet/context.rs]: I would leave the line though, I think it's better to be consistent xD

**Review by piotmag769 — APPROVED** [review]

File: crates/forge/tests/data/steps/src/lib.cairo [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    fn steps_570031() {
+        let mut i = 0;
+
+        loop {
+            if i == 30_000 {
+                break;
+            }
+
+            i = i + 1;
```
Comment (war-in) [inline crates/forge/tests/data/steps/src/lib.cairo:13]: [nit] We could probably perform some more complicated operations to reach the expected steps faster

**Review by war-in — APPROVED** [review]

**Pull request merged**